### ISSUE TITLE
Migration with no wallet imported from 1.4.1 fails

### ISSUE THREAD

**Issue by disassembler** [input-output-hk/cardano-wallet-legacy#230]
| Release | Operating System       | Cause |
| ---     | ---                    | ---   |
| next    | Windows & OSX & Linux) | Code  v Configuration v Environment v Human v Unknown |

# Context
 
Testing some unrelated changes on develop, I came across this error when trying to switch from 1.4 -> develop.

# Steps to Reproduce

```
git checkout tags/2.0
nix-build -A connectScripts.staging.wallet -o launch_staging_2.0
git checkout origin/develop
nix-build -A connectScripts.staging.wallet -o launch_staging_develop
./launch_staging_2.0
wait for sync
ctrl-c
./launch_staging_develop
```

## Expected behavior

node to start and function.

## Actual behavior
 
```
[cardano-sl.node:Error:ThreadId 5] [2019-01-16 20:20:44.87 UTC] logException: Could not parse saved checkpoint due to the following error: Failed reading: safecopy: Cardano.Wallet.Kernel.DB.Updates.Updates: Cannot find getter associated with this version number: Version {unVersion = 0}
From:	Cardano.Wallet.Kernel.DB.AcidState.DB:


CallStack (from HasCallStack):
  error, called at src/Data/Acid/Local.hs:329:7 in acid-state-0.15.0-6wvtuBINZDgEicddS27lrz:Data.Acid.Local
cardano-node: Could not parse saved checkpoint due to the following error: Failed reading: safecopy: Cardano.Wallet.Kernel.DB.Updates.Updates: Cannot find getter associated with this version number: Version {unVersion = 0}
From:	Cardano.Wallet.Kernel.DB.AcidState.DB:


CallStack (from HasCallStack):
  error, called at src/Data/Acid/Local.hs:329:7 in acid-state-0.15.0-6wvtuBINZDgEicddS27lrz:Data.Acid.Local
```
---

# Resolution Plan

After we split the wallet backend development on its own repo, we wanted
to setup a submodule from 'cardano-sl' to 'cardano-wallet' to keep the
former up-to-date with what was happening on 'cardano-wallet'. This was
unpractical to maintain and decided to revert this and instead,
duplicate and freeze the source code of 'cardano-wallet' on sl (only
updating when changes required so).

When we froze, it wasn't yet clear what features were going to be
released in 1.5. Therefore, the code currently on 'cardano-sl' includes
stuff it should so we have to remove a bit a few things that were introduced as part of the development.

Those introductions actually changed the `DB` representation cascading all sort of `acid-state` errors when trying to migrate the underlying data. 

# PR

| Number   | Base      |
| ---      | ---       |
| input-output-hk/cardano-sl#4020 | `develop`

# QA 

This was pretty "easy" to solve: just remove everything that shouldn't be there. Overall, it would be nice to do another round of sanity checka on `cardano-sl/develop` when we decide we are ready for the release. There are likely some other changes we do want to revert and postpone to further releases.

**Comment by piotr-iohk** [comment]
I have a few questions:

 - does this mean that what actually is released as wallet is https://github.com/input-output-hk/cardano-sl/tree/develop/wallet rather than https://github.com/input-output-hk/cardano-wallet. Is that what's going to happen in the future also?
 - are those changes are reverts of particular commits to have this copy of frozen wallet on a certain level, or explicitly removing parts of the code that should not be there (the latter seems to be potentially error prone)
 - what type of tests have been executed after those changes? In particular I do not see the integration tests as being the part of this frozen wallet (https://github.com/input-output-hk/cardano-sl/tree/develop/wallet/test/integration) and they are mainly testing existing features so, perhaps, it would be nice to have them executed against this wallet copy.

>There are likely some other changes we do want to revert and postpone to further releases.

Would it make sense to create a ticket for that already in order to have this activity it planned?

**Comment by KtorZ** [comment]
> does this mean that what actually is released as wallet is https://github.com/input-output-hk/cardano-sl/tree/develop/wallet rather than https://github.com/input-output-hk/cardano-wallet. Is that what's going to happen in the future also?

For 1.5, yes. Future releases will be pulled from `cardano-wallet`. 

> are those changes are reverts of particular commits to have this copy of frozen wallet on a certain level, or explicitly removing parts of the code that should not be there (the latter seems to be potentially error prone)

Well, it does revert commits that shouldn't be on the frozen `cardano-sl/wallet`. It also happens that we've also made some cleanup on `cardano-wallet` which overlap with this. So, some of this stuff was highly "experimental" and we've moved to another design already. The PR above brings `cardano-sl/wallet` back to before we started these experimentations. Note that, these were endpoints and logic not exposed to users and (mostly) separated from the core code. So, there's little risk in removing them. 

> what type of tests have been executed after those changes? In particular I do not see the integration tests as being the part of this frozen wallet (https://github.com/input-output-hk/cardano-sl/tree/develop/wallet/test/integration) and they are mainly testing existing features so, perhaps, it would be nice to have them executed against this wallet copy.

Integration tests aren't ran on `cardano-sl`, they were disabled some time ago already. For testing, I actually ran the existing suite + the reproduction steps detailed above. Everything went well once we removed the EOS wallets stuff for which we didn't defined any DB migration. 
The situation is clearly not ideal on `cardano-sl/develop` and that's also the main reason why we've frozen development there quite early on. We still have included patches and bugfixes as they were coming along.

> Would it make sense to create a ticket for that already in order to have this activity it planned?

Definitely!

**Comment by KtorZ** [comment]
@piotr-iohk: https://github.com/input-output-hk/cardano-wallet/issues/251

**Comment by piotr-iohk** [comment]
>For testing, I actually ran the existing suite...

OK, so you did run the existing test suite (including integration) against the `cardano-sl\wallet` locally. Is that correct?

**Comment by KtorZ** [comment]
> OK, so you did run the existing test suite (including integration) against the cardano-sl\wallet locally. Is that correct?

I ran what's available on `cardano-sl` :disappointed: ... So not the integration tests _yet_.

**Comment by piotr-iohk** [comment]
I would be strongly for running _our_ integration tests on that frozen wallet then if possible :)

There is 100 valid scenarios now, and that's already something. It would be a pity not to use them and then, for instance, find bugs during _regression_ phase that could have been potentially discovered with those tests much earlier...

**Comment by KtorZ** [comment]
@piotr-iohk agreed, that's what I've included in #251 as a requirement. And we won't put 1.5 into QA until #251 has been tackled.

**Comment by piotr-iohk** [comment]
Ok. missed that! Cool then :)
